### DISCUSSION TITLE
Only upper half of P5 2121 64x32 16S module work

### DISCUSSION THREAD

**Discussion by betopashic** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#439] [Q&A]
Hi there, I use ESP32-HUB75-MatrixPanel-DMA with ESP32 and P5 2121 64x32 16S module with ICN2038S.
Only upper half of module work. Can somebody advice for solution? Thanks, Regards
![1681638723106](https://user-images.githubusercontent.com/130899728/232291507-741dc33d-5b93-4696-a1d4-daa0c369e338.jpg)

**Comment by mrcodetastic** [comment]
Looking like you have a electrical wiring issue more than anything. All the LEDs shouldn't be glowing green in any case, especially if running the normal "bouncing squares" demo

**Reply by betopashic** [reply]
Thanks for quick reply, I reconnect all wire's, everything look ok. I test with all examples in lib, the same problem.
I will try to use another led panel.

**Reply by mrcodetastic** [reply]
Double check your grounding as well. You must make sure the ground on the ESP32 is going to one of the ground pins on the HUB75 connector as well.

**Reply by betopashic** [reply]
I check all, problem is still here. I have several led modules, with different pixel pitch, different scan mode, different IC led driver. But without result. Can you please specify some led module that you use and work ok with your lib? Pixel pitch, scan mode, IC led driver, so I will try to found that model. Thanks, regards.

**Reply by mrcodetastic** [reply]
Have you read the README?

**Reply by betopashic** [reply]
Yes of course. I read README file - led panel with ICN2038S, 16S, 64x32 pixels don't work. I try and check everything, no result.

Now, I use code from kucukkose on P6 (3535) 8S 32x32 led panel with TC5020AP IC led driver and the result is like on attached image.
With
`dma_display->setCursor(0,0); dma_display->print("Hello");`
text start on x=0 and y=9.

When I try with
`dma_display->setCursor(0,10); dma_display->print("Hello");`
the result is like on second image.

I try also with second solution from kucukkose with 1/8 Scan Panel Demonstration but a lot of error is here:
`OneEightMatrixDisplay = new VirtualMatrixPanel((*dma_display), NUM_ROWS, NUM_COLS, PANEL_RES_X, PANEL_RES_Y, SERPENT, TOPDOWN);`
with candidate expects 6 arguments, 7 provided explanation and error: 'ONE_EIGHT_32' was not declared in this scope 
I try with some lib modification but without result.

Advice for solution? Thanks and best regards.
![11](https://user-images.githubusercontent.com/130899728/233441642-10fae941-42bf-420e-ba1f-0b83ca4056b8.jpg)
![22](https://user-images.githubusercontent.com/130899728/233441676-91e84344-045b-4621-8bff-ab18faeb6f78.jpg)

**Reply by rigorka** [reply]
Looks like you might have a row swapping issue similar to the one I had with 64x64 1/16 panels.

Try
```
v_display = new VirtualMatrixPanel((*dma_display), NUM_ROWS, NUM_COLS, PANEL_RES_X, PANEL_RES_Y, CHAIN_TOP_RIGHT_DOWN);
    
// THE IMPORTANT BIT BELOW!
v_display->setPhysicalPanelScanRate(FOUR_SCAN_64PX_HIGH);
```

If that does not help - try to write a test code which lights up the pixels line by line with some 0.5 sec delay between the lines, from y=0 to y=31, and see in what order the real lines will be appearing at on the screen. This might give some clues on how to fix your issue.

**Reply by tansinbaby** [reply]
i have a stc11f32xe lqfp44 i have written a lot of code but my led panel is not working just got blank each time can anyone guide me or write c program for me

**Comment by Consokbd** [comment]
Bonjour, j'utilise un ESP32-HUB75-MatrixPanel-DMA avec un module ESP8266 et P5 2121 64x32 16S M5
j'ai respecté le câblage
![IMG-20250505-WA0006](https://github.com/user-attachments/assets/039e770e-974b-4c78-ac60-1ce14bd052cb)
![IMG-20250505-WA0007](https://github.com/user-attachments/assets/a5c2f572-a07a-408d-b9fd-212fb16e4a6c)
![IMG-20250505-WA0008](https://github.com/user-attachments/assets/594b1bbc-fbf4-4407-b9e5-b4caf1b0b8fd)
![IMG-20250505-WA0009](https://github.com/user-attachments/assets/4d230083-e806-46f8-b624-4faa84cfc830)
 et je test le code d'exemple pixel times mais l'écrans ne donne rien quel est le problème selon vous ?
aidez moi svp a résoudre le problème.

**Reply by mrcodetastic** [reply]
Bonjour. Je suis étonné que vous ayez pu compiler la bibliothèque avec l'ESP8266 ? Comment avez-vous fait ? Cette bibliothèque ne prend en charge qu les appareils ESP32. Êtes-vous sûr que c'est bien un ESP8266 ?

**Comment by Consokbd** [comment]
Bonjour j'ai essayer avec l'ESP8266 ça ne marcher pas 
alors j'ai utilisé un ESP32, les texte sont déformé et illisible, est ce que je peux avoir une solution svp ?

![WhatsApp Image 2025-05-06 à 15 07 15_b6b21058](https://github.com/user-attachments/assets/60189ce6-5a18-4dd0-88aa-ec4f0928b505)
